Simple Incremental/Assets/Scripts/Movement.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class Movement : MonoBehaviour
+{
+
+    public event MovementFinishedHandler OnMovementFinished;
+    public delegate void MovementFinishedHandler();
+
+    public float moveSpeed;
+    
+    [SerializeField] 
+    private Transform target;
+
+    [SerializeField] 
+    private float terminationDistance = 0.01f;
+
+    private bool shouldMove = true;
+    private bool movementFinished = false;
+   
+    public void StartMovement()
+    {
+        movementFinished = false;
+        shouldMove = true;
+    }
+    
+    public void StopMovement()
+    {
+        shouldMove = false;
+    }
+
+    public void Update()
+    {
+        if (!movementFinished && shouldMove)
+        {
+            if (Vector3.Distance(target.position, transform.position) > terminationDistance)
+            {
+                transform.position =
+                    Vector3.MoveTowards(transform.position, target.position, Time.deltaTime * moveSpeed);
+            }
+            else
+            {
+                movementFinished = true;
+                OnMovementFinished?.Invoke();

```

Comment:
If other scripts need to know when the enemy is done moving, this can stay. Otherwise, consider removing the event since no one is expecting to consume it.

If you foresee this event being used to drive other game logic, consider using a scriptableObject GameEvent here instead (see Milestone 4 for an example)

Reply:
Since your reasoning for the movement is sound, you can leave this with delegates/events. I agree with your implementation rather than using scriptable objects.